rust/rope/Cargo.toml

```diff
@@ -4,7 +4,7 @@ description = "A generic rope data structure built on top of B-Trees."
 license = "Apache-2.0"
 name = "xi-rope"
 repository = "https://github.com/google/xi-editor"
-version = "0.2.0"
+version = "0.2.1"
```

Comment:
If we're bumping, probably to 0.3. I think this is not semver-compatible.

Reply:
My thinking here was just that I wanted to use the `#[deprecated]` annotation on the old Interval constructors, and that wants to know the version the deprecation occurred. I think with semver, pre 1.0, anything goes, although there may be some unofficial conventions we're violating.

Reply:
No, it's not "anything goes," 0.x.y to 0.x.(y+1) is supposed to not break code, but 0.x to 0.(x+1) is allowed to (this is slightly different than the original semver, as I recall).

Reply:
Cargo docs mention that official semver does not actually specify 0.x.y as a special case but that cargo among others considers any 0.x.y version bumps to be breaking changes.

Reply:
It's also unclear that this breaks any code; at least it doesn't break API? The only API breaking change here is that I made `push_subseq` `pub(crate)` from `pub` because I thought it probably shouldn't be public API, but that's really out of scope anyway.

In either case, happy to defer to the crowd. I was thinking of doing a 3.0 of the rope crate shortly (and publishing to crates.io) but I wanted to finish this current batch of cleanup first.

Reply:
oops no it does remove some public methods from `Interval`, so yea, breaking.